vmware_dvs/agent/firewalls/vcenter_firewall.py

```diff
@@ -31,19 +32,33 @@
 CLEANUP_REMOVE_TASKS_TIMEDELTA = 60
 
 
-def firewall_updater_loop(list_queues, remove_queue):
-    pq = PortQueue(list_queues, remove_queue)
-    pq.port_updater_loop()
-    while True:
-        r_ports = pq.get_remove_tasks()
-        if r_ports:
-            remover(pq, r_ports)
+def firewall_main(list_queues, remove_queue):
+    dvs_firewall = dvs_firewall_updater(list_queues, remove_queue)
+    signal.signal(signal.SIGTERM, dvs_firewall._handle_sigterm)
+    dvs_firewall.updater_loop()
 
-        dvs, ports = pq.get_update_tasks()
-        if dvs and ports > 0:
-            updater(dvs, ports)
-        else:
-            time.sleep(1)
+
+class dvs_firewall_updater(object):
```

Comment:
Accordingly PEP, please, use CamelCase for class names: dvs_firewall_updater -> DVSFirewallUpdater